### PR TITLE
fix: bug in guttertooltip when `tooltipsFollowsMouse` set to false

### PR DESCRIPTION
**Pull request by akoreman**
*Issue #, if available:* NA

Fixes a bug where an error is thrown when hovering over the gutter with `tooltipsFollowsMouse == true` and scrolled. Additionally, adds `tooltipsFollowsMouse` to kitchen-sink to make it easier to test this option.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/layer/lines.js [hunk 1]
```diff
@@ -42,6 +42,15 @@ class Lines {
     get(index) {
         return this.cells[index];
     }
+
+    getCellAtRow(row) {
+        for (var i = 0; i < this.getLength(); i++){
```
Comment (whazor) [inline src/layer/lines.js:47]: this could be a binary search right? or maybe something faster

Comment (akoreman) [reply src/layer/lines.js]: changed the implementation of the search

File: src/mouse/mouse_event.js [hunk 2]
```diff
@@ -48,6 +48,18 @@ class MouseEvent {
...
         return this.$pos;
     }
+
+    /**
+     * Get the relative position within the gutter.
+     * 
+     * @return {Number} 'row' within the gutter. 
+     */
+    getGutterRow() {
+        var documentRow = this.editor.renderer.screenToTextCoordinates(this.clientX, this.clientY).row;
```
Comment (ibabanov-aws) [inline src/mouse/mouse_event.js:58]: Might be better to use `this.getDocumentPosition().row` here

**Review by ibabanov-aws — APPROVED** [review]

File: src/mouse/mouse_event.js (hunk 2 above)
Comment (akoreman) [reply src/mouse/mouse_event.js]: ah yeah, good catch, changed

File: src/layer/lines.js (hunk 1 above)
Comment (akoreman) [reply src/layer/lines.js]: implementation is changed to not use any search anymore

**Review by ibabanov-aws — APPROVED** [review]

**Pull request merged**